examples/apps/table-detail-tea/src/main.rs

```diff
@@ -0,0 +1,239 @@
+            data.name.as_str(),
+            data.address.as_str(),
+            data.email.as_str(),
+        ])
+        .height(2)
+    });
+
+    let table = Table::new(
+        rows,
+        [
+            Constraint::Length(20),
+            Constraint::Min(10),
+            Constraint::Fill(1),
+        ],
+    )
+    .header(header)
+    .row_highlight_style(Style::default().add_modifier(Modifier::REVERSED | Modifier::ITALIC))
+    .block(
+        Block::default()
+            .borders(Borders::ALL)
+            .title(" Terminal fans "),
+    );
+
+    frame.render_stateful_widget(table, top, &mut model.table_state);
+
+    // Select the first row if no row is selected
+    if model.table_state.selected().is_none() {
+        model.table_state.select_first();
+    }
+
+    // Detail view
+    let selected_item = &model.table_items[model.table_state.selected().unwrap()];
+    let detail = Paragraph::new(format!(
+        "{}\n\n{}",
+        selected_item.email, selected_item.address
+    ))
+    .block(
+        Block::default()
+            .borders(Borders::ALL)
+            .title(format!(" {} ", selected_item.name))
+            .title_bottom(Line::from(INFO_TEXT))
+            .padding(Padding::new(1, 1, 1, 1)),
+    );
+    frame.render_widget(detail, bottom);
+}
+
+fn handle_event(_: &Model) -> color_eyre::Result<Option<Message>> {
+    if event::poll(Duration::from_millis(250))? {
+        match event::read()? {
+            Event::Key(key) if key.kind == event::KeyEventKind::Press => Ok(handle_key(key)),
+            Event::Mouse(mouse) => Ok(handle_mouse(mouse)),
+            _ => Ok(None),
+        }
+    } else {
+        Ok(None)
+    }
+}
+
+const fn handle_key(key: event::KeyEvent) -> Option<Message> {
+    match key.code {
+        KeyCode::Char('j') | KeyCode::Down => Some(Message::SelectNext),
+        KeyCode::Char('k') | KeyCode::Up => Some(Message::SelectPrevious),
+        KeyCode::Char('q') | KeyCode::Esc => Some(Message::Quit),
+        _ => None,
+    }
+}
+
+const fn handle_mouse(mouse: event::MouseEvent) -> Option<Message> {
+    match mouse.kind {
+        MouseEventKind::ScrollDown => Some(Message::SelectPrevious),
+        MouseEventKind::ScrollUp => Some(Message::SelectNext),
+        _ => None,
+    }
+}
+
+fn update(model: &mut Model, msg: Message) -> Option<Message> {
+    match msg {
+        Message::Quit => model.running_state = RunningState::Done,
+        Message::SelectNext => {
+            let current_index = model.table_state.selected().unwrap_or(0);
+            if current_index < model.table_items.len() - 1 {
+                model.table_state.select(Some(current_index + 1));
+            }
+        }
+        Message::SelectPrevious => {
+            let current_index = model.table_state.selected().unwrap_or(0);
+            if current_index > 0 {
+                model.table_state.select(Some(current_index - 1));
+            }
+        }
+    };
+    None
+}
+
+mod tui {
```

Comment:
Mostly this would be dropped - see the comment further up

Reply:
Verbatim copy from the manual. Please adjust as you see fit.